src/Microsoft.AspNet.Mvc.ModelBinding/IObjectModelValidator.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.ModelBinding;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// A validator that validates a given object and adds erros to the given <see cref="ModelStateDictionary"/>.
+    /// </summary>
+    public interface IObjectModelValidator
```

Comment:
Suggestion: 
- `IModelValidator` for the top-level object
- `IPropertyValidator` for the property-level extensibility